panels/user-accounts/um-utils.c

```diff
@@ -460,24 +460,8 @@ down_arrow (GtkStyleContext *context,
         cairo_restore (cr);
 }
 
-static guint
-get_login_name_max (void)
-{
-#ifdef LOGIN_NAME_MAX
-        return LOGIN_NAME_MAX;
-#else
-        static gint length;
-
-        if (!length) {
-                length = sysconf (_SC_LOGIN_NAME_MAX);
-                g_assert_cmpint (length, >=, 0);
-        }
-
-        return length;
-#endif
-}
-
-#define MAXNAMELEN  get_login_name_max ()
+/* This is best hardcoded, see GNOME #766401. */
+#define MAXNAMELEN  32
```

Comment:
I think it's OK to deal with this in a separate issue, but it really needs to be fixed in the same release, else it will be very hard to figure out why user account creation is failing. It seems easiest to track it in the same issue.

I don't know how to respond to Matthias's review, so I filed a bug against shadow-utils. Let's see if they respond and what they think.